src/Features/SupportQueryString/BaseUrl.php

```diff
@@ -94,7 +121,13 @@ public function urlName()
     public function getFromUrlQueryString($name, $default = null)
     {
         if (! app('livewire')->isLivewireRequest()) {
-            return request()->query($this->urlName(), $default);
+            $value = request()->query($this->urlName(), $default);
+
+            if (array_key_exists($name, request()->query()) && $value === $default) {
```

Comment:
I don't think this logic should live here for two reasons:

1) It doesn't account for "lazy" components that don't get handled in this condition, but get handled in `getFromRefererUrlQueryString` further down.

2) Structurally, I wonder if this logic can be more closely located to the line 73 additions.

In general, changes like this that affect multiple parts of a file like this, should ideally be refactored to live together, or be well commented so people know why those conditions are added.

So at least, please add comments to describe why that logic is there.

Reply:
> Structurally, I wonder if this logic can be more closely located to the line 73 additions.

That is not possible because then the code would be stopped at line 59 (`if ($initialValue === 'noexist') return;`) 

I added a comment to describe why that code exists there

Reply:
Perfect explanations. Thank you!